app/models/category.py

```diff
@@ -1,102 +1,122 @@
+
+
+# ✅ Progress 테이블: 카테고리별 목표 진행률 저장 테이블
+class Progress(Base):
+    __tablename__ = "progresses"
+
+    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
+    category_id: Mapped[UUID] = mapped_column(
+        ForeignKey("categories.id"), nullable=False
     )
-    monthly_achievements: Mapped[List["app.models.evaluation.MonthlyAchievement"]] = (
-        relationship(
-            "app.models.evaluation.MonthlyAchievement",
-            back_populates="category_progress",
-            cascade="all, delete",
-        )
+    progress_rate: Mapped[Optional[DECIMAL]] = mapped_column(
+        DECIMAL(5, 2), default=0.00
     )
+    start_date: Mapped[Date] = mapped_column(Date, nullable=False)
+    end_date: Mapped[Date] = mapped_column(Date, nullable=False)
 
+    # 관계: Progress N : 1 User
+    user: Mapped["app.models.user.User"] = relationship(
+        "app.models.user.User",
+        back_populates="progresses",
+        foreign_keys="Progress.user_id",
+    )
+    # 관계: Progress 1 : 1 Category
+    category: Mapped["app.models.category.Category"] = relationship(
+        "app.models.category.Category",
+        back_populates="progress",
+        foreign_keys="Progress.category_id",
+    )
 
-# 목표 관리 테이블 (Goal) (여러 목표가 한 카테고리에 연결됨)
-class Goal(Base):
-    __tablename__ = "goals"
 
-    progress_id: Mapped[UUID] = mapped_column(
-        ForeignKey("category_progresses.id"), nullable=False
-    )
+# ✅ Todo 테이블: 할 일(Task) 기록 테이블
+class Todo(Base):
+    __tablename__ = "todos"
 
     user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
-    category_name: Mapped[CategoryNameEnum] = mapped_column(
-        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
+    category_id: Mapped[UUID] = mapped_column(
+        ForeignKey("categories.id"), nullable=False
     )
-    category_color: Mapped[CategoryColorEnum] = mapped_column(
-        Enum(CategoryColorEnum, name="category_color_enums"), nullable=False
-    )
-    todo_content: Mapped[str] = mapped_column(Text, nullable=False)
+    week_id: Mapped[int] = mapped_column(ForeignKey("weeks.id"), nullable=False)
+    content: Mapped[str] = mapped_column(Text, nullable=False)
     start_date: Mapped[Date] = mapped_column(Date, nullable=False)
     end_date: Mapped[Date] = mapped_column(Date, nullable=False)
-    is_completed: Mapped[bool] = mapped_column(Boolean, default=False)
-    is_repeat: Mapped[bool] = mapped_column(Boolean, nullable=False)
+    is_completed: Mapped[Optional[bool]] = mapped_column(Boolean, default=False)
+    is_repeat: Mapped[Optional[bool]] = mapped_column(Boolean, default=False)
 
-    user: Mapped["User"] = relationship("User", back_populates="goals")
-    category_progress: Mapped["CategoryProgress"] = relationship(
-        "CategoryProgress", back_populates="goals"
+    # 관계: Todo N : 1 User / Category
+    user: Mapped["app.models.user.User"] = relationship(
```

Comment:
여긴 `Optional`이 없어도 되나요?

```suggestion
    user: Mapped[Optional["app.models.user.User"]] = relationship(
```

Reply:
@GideokKim 
유저는 최상위 테이블로 다른 테이블이 만들어질 시점에는 유저가 필수로 있다고 생각하여 Optional을 추가하지 않은것으로 보입니다.

Reply:
유저는 무조건 값이 있고 없으면 안되는 값이라 Optional필수로 넣었습니다